test/integration/local/message-service.test.js

```diff
@@ -11,41 +15,50 @@ const generateSampleClaim = () => ({
 })
 
 describe.only('Test message service', () => {
+  let messageReceiver
   let messageService
+  let message
+  const testConfig = { ...config.messageQueues.claimQueue }
 
   beforeAll(async () => {
-    messageService = await require('../../../server/services/message-service')()
     await asbHelper.clearAllQueues()
-  }, 30000)
+    message = generateSampleClaim()
+    messageService = await MessageService()
 
-  beforeEach(async () => {
-    await dbHelper.truncate()
-    jest.restoreAllMocks()
-  })
+    const messageSender = new MessageSender('message-service-sender-test', testConfig)
+    await messageSender.sendMessage(message)
+  }, 30000)
 
   afterAll(async () => {
+    await messageService.closeConnections()
+    await messageReceiver.closeConnections()
     await asbHelper.clearAllQueues()
     await dbHelper.close()
-    await messageService.closeConnections()
   }, 30000)
 
-  test('Message service sends the claim to schedule queue', async () => {
-    const message = generateSampleClaim()
-    const scheduleSender = messageService.scheduleSender
-    const spy = jest.spyOn(scheduleSender, 'sendMessage')
+  test('Processed message ends up on scheduleQueue', async () => {
+    let done
+    const promise = new Promise((resolve) => {
+      done = resolve
+    })
+    const action = (result) => {
+      done(result.hello === message.hello)
+    }
 
-    await messageService.publishClaim(message)
-    await expect(spy).toHaveBeenCalledTimes(1)
-    await expect(spy).toHaveBeenCalledWith(message)
+    messageReceiver = new MessageReceiver('test-receiver', { ...config.messageQueues.scheduleQueue }, undefined, action)
+    return expect(promise).resolves.toEqual(true)
   })
 
-  test('Message service sends the claim to calculation queue', async () => {
-    const message = generateSampleClaim()
-    const calculationSender = messageService.calculationSender
-    const spy = jest.spyOn(calculationSender, 'sendMessage')
+  test('Processed message ends up on calculationQueue', async () => {
+    let done
+    const promise = new Promise((resolve) => {
+      done = resolve
+    })
+    const action = (result) => {
+      done(result.hello === message.hello)
```

Comment:
similar to the above, `hello` will always be `undefined`

Reply:
You can't beat a test for `undefined === undefined`! 😞